### PR TITLE
Online indicator

### PR DESCRIPTION
**Pull request by Mikunj**
I couldn't really test this since lokinet doesn't work on Mac at the moment.
screen shot 2019-02-01 at 1 12 00 pm
screen shot 2019-02-01 at 1 12 11 pm
Currently this is only shown on the left menu, do we want to show them on the conversation header as well?

### CODE REVIEW

File: js/conversation_controller.js [hunk 1]
```diff
@@ -263,13 +270,14 @@
...
-              conversation.updateLastMessage(),
-              conversation.updateProfile(),
-              conversation.updateProfileAvatar(),
-              conversation.resetPendingSend(),
-              conversation.updateProfile(),
-            ]);
+            promises.push(async () => {
+              // We need to await each one as
+              //  we don't want to simultaneously call writes on the sql database
+              await conversation.updateLastMessage();
```
[the remark this answers is not in the document]
Comment (BeaudanBrown) [reply js/conversation_controller.js]: Seems correct to me

File: js/conversation_controller.js [hunk 2]
```diff
@@ -248,6 +248,13 @@
     async load() {
       window.log.info('ConversationController: starting initial fetch');
 
+      // We setup online and offline listeners here because we want
+      //  to minimize the amount of listeners we have to avoid memory leaks
+      if (!this.p2pListenersSet) {
```
Comment (sachaaaaa) [inline js/conversation_controller.js]: where is `this. p2pListenersSet` set?

Comment (Mikunj) [reply js/conversation_controller.js]: good catch, i forgot about it

File: js/modules/data.js [hunk 3]
```diff
@@ -688,11 +689,13 @@ async function getConversationCount() {
 }
 
 async function saveConversation(data) {
-  await channels.saveConversation(data);
+  const cleaned = omit(data, 'isOnine');
```
Comment (sachaaaaa) [inline js/modules/data.js]:
```suggestion
  const cleaned = omit(data, 'isOnline');
```

**Pull request merged**